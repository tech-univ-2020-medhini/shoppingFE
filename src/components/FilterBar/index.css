.filter-bar{
    display:flex;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    padding: 0rem;
    justify-content: center;
    height: 2rem;
}
.category-button{
    display:flex;
    flex:1;
    background-color: rgb(196,196,196);
    height: 2rem;
    width: 4rem;
}