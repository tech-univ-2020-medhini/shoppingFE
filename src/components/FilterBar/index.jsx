import React from 'react';
import PropTypes from 'prop-types';
import './index.css';


function FilterBar(props) {
  const { categoryList, onCLickFilter } = props;
  const categories = [<button type="button" className="category-button" onClick={() => onCLickFilter('All')}>All</button>];
  if (categoryList !== undefined) {
    categoryList.forEach((category) => {
      categories.push(<button type="button" className="category-button" onClick={() => onCLickFilter(category)}>{category}</button>);
    });
  }

  return (
    <div className="filter-bar">
      {categories}
    </div>

  );
}
FilterBar.propTypes = {
  categoryList: PropTypes.arrayOf(PropTypes.string).isRequired,
  onCLickFilter: PropTypes.func.isRequired,
};

export default FilterBar;
