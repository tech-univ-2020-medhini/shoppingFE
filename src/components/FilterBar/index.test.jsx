import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FilterBar from './index';


describe('the product cards', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<FilterBar />);

    expect(asFragment()).toMatchSnapshot();
  });
});
