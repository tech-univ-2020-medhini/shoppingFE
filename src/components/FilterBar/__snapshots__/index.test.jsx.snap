// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the product cards should render correctly 1`] = `
<DocumentFragment>
  <div
    class="filter-bar"
  >
    <button
      class="category-button"
      data-testid="filter-btn"
      type="button"
    >
      All
    </button>
  </div>
</DocumentFragment>
`;
